packages/backend-storage/src/storage_container_entry_generator.ts

```diff
@@ -30,7 +31,9 @@ export class StorageContainerEntryGenerator
     scope,
     ssmEnvironmentEntriesGenerator,
   }: GenerateContainerEntryProps) => {
-    const amplifyStorage = new AmplifyStorage(scope, this.props.name, {
+    validateResourceName(this.props.name);
+    const sanitizedName = this.sanitizeName(this.props.name);
+    const amplifyStorage = new AmplifyStorage(scope, sanitizedName, {
```

Comment:
can we try just passing the validated name as the construct id? I think CDK and CFN will handle generating valid logical IDs and physical IDs for us.

Reply:
we allow spaces in the validated name. Don't think cdk/cfn does any sanitation for that https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.CfnBucket.html#bucketname

Reply:
Just tested out
```ts
    const amplifyStorage = new AmplifyStorage(scope, 'TEST name_With-spaces', {
```
and it worked fine. The logical ID for the bucket became `TESTnameWithspacesBucketFF5F524C` and the bucket physical id was `amplify-expand-foyleef-sa-testnamewithspacesbucket-5ufwoaq6g09w`

Reply:
gotcha, we are not setting the bucketName prop and letting CFN create one from logicalId. Will remove this.